src/foam/dao/jdbc/JDBCPooledDataSource.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.dao.jdbc;
+
+import foam.core.X;
+import org.apache.commons.dbcp2.*;
+import org.apache.commons.pool2.ObjectPool;
+import org.apache.commons.pool2.impl.GenericObjectPool;
+
+import foam.nanos.logger.Logger;
+
+import javax.sql.DataSource;
+import java.sql.DriverManager;
+
+public class JDBCPooledDataSource {
+
+  // Holds a reference to the connection pool ( .getConnection() )
+  protected DataSource dataSource_;
+  protected String poolName_;
+
+  public JDBCPooledDataSource(X x, String poolName) {
```

Comment:
Why do we need to supply the poolName?

Reply:
So that the DAO chooses which pool to use, so we can have different DAOs use different pools if we want to have a DAO with high priority use its own exclusive pool